### ISSUE TITLE
Is it possible to detect where the collision occurred on the car/bounding box? I am able to detect collisions but would like to check to some level of certainty (say clock points) where the collision has occurred? 

### ISSUE THREAD

**Issue by pjk19942** [esmini/esmini#483]

**Comment by eknabevcc** [comment]
Unfortunately there is no automatic way to get more information from a collision, other than it happened and with what object.

However, you can utilize replayer to inspect the collision and get some more information. Try this:

`./bin/esmini --headless --osc ./resources/xosc/ltap-od.xosc --disable_controllers --fixed_timestep 0.033 --record sim.dat`

`./bin/replayer --window 60 60 800 400 --file sim.dat --res_path ./resources --collision`

It will pause as soon there is a collision and output something like:

```
Collision between 0 and 1 at time 6.57.
- Relative speed 10.80 km/h
- Angle -210.08 degrees (ego to target)
```

This process is also described in [User guide - Replay scenario](https://esmini.github.io/#_replay_scenario).

BTW: Press ',' (or add launch argument `--view_mode boundingbox` to the launch arguments) to view bounding boxes.

**Comment by pjk19942** [comment]
Thank you for your reply, i believe there is an alternate issue here, in my case I have 3 vehicles, 2 cars and 1 semi truck trailer. For the semi truck i am using  xosc.CatalogReference('VehicleCatalog', 'semi_truck') . During the simulation the Semi-truck appears as V3,V3+ (The trailer) . I can still ignore it however in the collision script that you shared 
Collision between 0 and 1 at time 0.03.
- Relative speed 104.47 km/h
- Angle 0.00 degrees (ego to target)

despite continuing the simulation and 2 more collisions being occurred the output is just shown as the above (only 1 instance collision) . Which makes me believe the replayer detects the first collision , maybe, but since the truck and trailer are attached i believe its misidentifying that as a collision. Interestingly the log file doesnt show this.

**Comment by eknabevcc** [comment]
I tried to reproduce as follows: One car driving trough two stand still vehicles: First a semi_truck and then another car. 

I run:
`./bin/esmini --window 60 60 800 400 --osc ./resources/tmp/collision.xosc --fixed_timestep 0.01 --record sim.dat`
`./bin/replayer --window 60 60 800 400 --res_path ./resources --file sim.dat --collision`

and get the two expected collisions:
```
Collision between 0 and 2 at time 0.70.
- Relative speed 36.00 km/h
- Angle 0.00 degrees (ego to target)
Collision between 0 and 3 at time 3.50.
- Relative speed 36.00 km/h
- Angle 0.00 degrees (ego to target)
```
![collision](https://github.com/esmini/esmini/assets/29146839/82961ea4-a96c-4a9f-89b6-2514f6054fdd)

Scenariofile: [collision.zip](https://github.com/esmini/esmini/files/13060925/collision.zip)

BTW: replayer will pause at each collision. To progress the replay you have to press 'space' key (see [User guide - replayer command reference](https://esmini.github.io/#_replayer) for more info).

If you still have an issue, please share the scenario and we'll have a look 👍

**Comment by pjk19942** [comment]
[Collision_Example.zip](https://github.com/esmini/esmini/files/13220215/Collision_Example.zip)
Hi, I tried running your example and it works as you displayed, but trying it on my example it fails. I have attached my example. I have used the space key too but i still do not see the info for the subsequent crashes.

**Comment by eknabevcc** [comment]
Sorry, I must have missed your latest update, with the provided Collision_Example. 

I see you closed it now. Does it mean that the issue somehow is solved, or that you got tired waiting for answer? :)

**Comment by pjk19942** [comment]
Hi Emil,

Thanks, I was waiting and then we figured out an alternate solution. :) by extracting the trajectory and boundingbox information and tracing its Angle of impact. If there is any property within esmini that can log such information or can be applied here, kindly do let me know.

Thanks

**Comment by eknabevcc** [comment]
Indeed the were some shortcoming, or limitations, in the implementation. For example, only Ego (id 0) was checked and only one overlap registered at a time, so if multiple objects run into id 0, only the first is registered.

Those shortcomings should now be fixed in [v2.37.4](https://github.com/esmini/esmini/releases/tag/v2.37.4).
